### ISSUE TITLE
Set nocscopetag doesn't work?

### ISSUE THREAD

**Issue by jiji879** [shawncplus/phpcomplete.vim#110]
For example `1tag xxxx` always jump to itself if contains the same definition, details see in last comment of #58 .

**Comment by complex857** [comment]
Please excuse my ignorance, but could you expand on what seems to be problem?
I've read comment in #58, however those numbers are not chosen by random but by trying to match the file associated on the returned list with the searched symbol's file (this piece of info is from ` phpcomplete#LocateSymbol`).

I believe the #58 was there because with the `cscopetags` enabled the `:tag` command was ignoring the leading `[count]` in there.
The fact the same prefix number means different tags for invocation to invocation in itself isn't a bug or issue as far as i understand - please correct me if I'm wrong here.

**Comment by jiji879** [comment]
Thx. you know exactly what i mean. but doesn't `set nocscopetag` turn it off?

i understand that  the fixed tag_position will be found in same symbol right? so if a invocation be found in first of the tags: `1tag xxx`, it always jump to itself but not the actually definition. 

it seems like the jump is relative to tag of current file. i had tried `set nocst` or `set csto=1` but doesn't work.

**Comment by complex857** [comment]
Yes, as far as I know it does. I'm sorry, but I think I'm still at lost on what is broken exactly.

The `tag_position` number is always re-calculated for every invocation of the plugin. So the numbers we gonna stick in front of the `tag <symbol>` commands should always be recalculated. I don't think the `1tag ...` will always be "itself" the symbol you are on might not even be on the taglist. The ordering of the list should be done according [tag-priority](http://vimdoc.sourceforge.net/htmldoc/tagsrch.html#tag-priority). So if there's a symbol you are trying to jump is in the same file as a tag and the plugin thinks that in the current context (based on for example the guessed type of the variable in front of the tag if it's a method invocation or such) then yes, otherwise it will jump without a number and let the tag-priority decide (as if the plugin wasn't installed).

I'm still at lost on if you have some "This is what I do, This is what I expect to happen, This happens instead" kinda bug we can try to fix, if you could phrase your issue that way it would help me out a lot!

**Comment by jiji879** [comment]
ok, sorry for making you confused. 

The issue is: if the position of function `foo` be found in first of the `tag_list_of_foo` (tag_position = 1 which calculated in JumpToDefinition#1019) and current file has the same function `foo` (or tag, which means this tag is also in the`tag_list_of_foo`), it will always jump to the tag of current file but not the actually definition.

(`tag_list_of_foo` is a set of tags with same definition of `foo`)
